internal/controller/cloudsyncmgr/cloudsync.go

```diff
@@ -65,37 +65,49 @@ func (c *CloudSyncImpl) InitiateCloudSync(isCloudSet string) (err error) {
 		if strings.Compare(strings.ToLower(isCloudSet), "true") == 0 {
 			log.Println("CloudSync init set")
 			isCloudSyncSet = true
+			//Intialize the client and hashmap storing client data
+			mqttmgr.SetClient(nil)
+			mqttmgr.InitClientData()
 		}
 	}
 	return nil
 }
 
-//StartCloudSync is used to start the sync by connecting to the broker
-func (c *CloudSyncImpl) StartCloudSync(host string) (err error) {
-	if isCloudSyncSet && len(host) > 0 {
-		log.Info("Starting the CLoudsync Mgr")
-		go mqttmgr.StartMQTTClient(host)
-	}
-	return
-}
-
 // RequestCloudSyncConf is  configuration request handler
-func (c *CloudSyncImpl) RequestCloudSyncConf(message mqttmgr.Message, topic string, clientID string) string {
+func (c *CloudSyncImpl) RequestCloudSyncConf(host string, clientID string, message mqttmgr.Message, topic string) string {
 	log.Info(logPrefix, "Publishing the data to the cloud")
-	mqttClient = mqttmgr.GetClient()
-	mqttmgr.SetClientID(clientID)
 	resp := ""
-	if mqttClient.IsConnected() {
+	var wg sync.WaitGroup
+	if !isCloudSyncSet {
+		resp = "CloudSync is not Active. Please stop the container and rerun the container with cloudsync set"
+		return resp
+	}
+	if len(host) == 0 {
+		return "No broker host defined"
+	}
+	wg.Add(1)
+	errs := make(chan string, 1)
+	go func() {
+		errs <- mqttmgr.StartMQTTClient(host, clientID)
+		resp = <-errs
+		wg.Done()
+
+	}()
+	wg.Wait()
+	if resp != "" {
+		errresp := fmt.Sprintf("Error Connecting MQTT -> %s", resp)
+		return errresp
+	}
+
+	mqttClient = mqttmgr.GetClient()
```

Comment:
What about using a local variable for Client struct?
```suggestion
	mqttClient := mqttmgr.GetClient(clientID)
```

I think you can modify GetClient() to get Client struct from the `clientData` map.

Reply:
yes, mqttClient was used as shared variable in the initial design. Now with each connection having its own now its not required.

Reply:
Done!!